### PR TITLE
samples: net: Solved the payload issue to coap GET Method

### PR DESCRIPTION
**Pull request by tarunkum**
In sample application of COAP put and post
methods are modified in coap-client.c

Signed-off-by: Tarun Kumar Agarwal <tarun.kumar.agarwal@intel.com>

### CODE REVIEW

**Review by lpereira — CHANGES_REQUESTED** [review]
Patch LGTM, but please tell more about the issue that's being fixed in the commit message.  A good place to start is reading https://chris.beams.io/posts/git-commit/

**Comment by jukkar** [comment]
> In COAP for get method there is no need of payload so that it is removed for GET method. It is only required for POST & PUT method.

You could add this sentence to commit message so the it would tell "why" you did the change, current commit message does not tell that.

**Comment by jukkar** [comment]
Please change also the commit subject to read "samples: net: coap: Solved the payload issue in GET Method" so it is clear which component is being changed.

File: samples/net/coap_client/src/coap-client.c [hunk 1]
```diff
@@ -285,18 +285,17 @@ static void send_coap_request(u8_t method)
...
 		break;
 
 	case COAP_METHOD_PUT:
+		r = coap_packet_append_payload_marker(&request);
+		if (r) {
+			net_pkt_unref(pkt);
+			printk("Unable to append payload marker\n");
+			return;
+		}
 		r = coap_packet_append_payload(&request, (u8_t *)put_payload,
```
Comment (jukkar) [inline samples/net/coap_client/src/coap-client.c]: Please put empty line between } and r=...

File: samples/net/coap_client/src/coap-client.c [hunk 2]
```diff
@@ -307,6 +306,12 @@ static void send_coap_request(u8_t method)
 		break;
 
 	case COAP_METHOD_POST:
+		r = coap_packet_append_payload_marker(&request);
+		if (r) {
+			net_pkt_unref(pkt);
+			printk("Unable to append payload marker\n");
+			return;
+		}
 		r = coap_packet_append_payload(&request, (u8_t *)post_payload,
```
Comment (jukkar) [inline samples/net/coap_client/src/coap-client.c]: Same here

**Review by jukkar — CHANGES_REQUESTED** [review]

**Comment by tarunkum** [comment]
@jukkar 

sure Jukka I will change it now only.

**Review by rbtchc — COMMENTED** [review]
LGTM. A marker is required only when payload presents.
RFC 7252, section 3
"The presence of a marker followed by a zero-length payload MUST be processed as a message format error."

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by jukkar** [comment]
@lpereira Does this look ok to you now?

**Review by lpereira — APPROVED** [review]

**Comment by lpereira** [comment]
LGTM

**Pull request merged**